.gitlab/source_test/macos.yml

```diff
@@ -61,7 +61,9 @@ tests_macos:
     - inv -e rtloader.install
 
 .lint_macos_gitlab:
-  stage: source_test
+  # We can't move the lint jobs in the .gitlab/lint directory because we need to extend the .macos_gitlab section
```

Comment:
Why is this a problem? You can extend jobs that are not defined in the same file, as long as you include the file that contains it.

You could have a `.gitlab/common/macos.yml` file that contains `.macos_gitlab`, and include it here and in `.gitlab/lint/macos.yml`.

Reply:
You're right it works like that, fixed in [22bcf81](https://github.com/DataDog/datadog-agent/pull/29790/commits/22bcf8181e4d80da28137dc756d097ae58f8d62d)